### ISSUE TITLE
Minimal hardware config to test frappe/erpnext

### ISSUE THREAD

**Issue by charleslcso** [frappe/erpnext#31673]
### Information about bug

Hello,

We have been stress testing frappe/erpnext v13.21.0 for quite a while now. Mainly use it as an API server, and see if it is suitable for production.

We use python 3.10, supervisor, nginx, mariadb on 1GB RAM with 1 CPU. 

It runs pretty decently on fresh boot. As we develop (creating DocType, Docs, changing DocFields, using Notifications, sending emails, logging into Desk) it becomes very very slow. 
Any API GET, POST will take a long time, and most of them will fail too.

We created 7 new DocType. Each DocType has about 20 fields. We have 1 Notification. 5 new Users accounts are created. 

I know this hardware is not suitable at all.

But the question is, even I give frappe/erpnext 32GB RAM, 10 core CPU, would it eventually slow to a crawl too?

I want to get some idea on how to keep frappe/erpnext fresh when running on bare metal, always-on servers.

### Module

other

### Version

frappe v13.21.0
erpnext v13.21.0

### Installation method

manual install

**Comment by MaxMorais** [comment]
Hi @charleslcso!

I do have some instances running with minimal hardware dependencies!

For example, I got the system working on 512Mb Raspberry's
Or VPC's 1gb, 1vcpu, but, due the amount of resources taken by frappe it's challenging!

1 important think to note:

IO is they key

On a 512 Memory raspbery pi, processes was lagging due I/O, so I moved the installation to an SSD NVMe, but still the processor is bad!

The second thing I noticed was, removing the Database (Putting it in another server will boost the performance, since the big part of the IO is due the database writes)

So, after analyse all the I/O and find what should be moved, and how to move!
I found that is possible to run ERPNext in inexpensive hardware requiments, only if you deal with the I/O.

The speed never will be the best compared to a baremetal server, but do a pretty decent job, for about 2/USD month!
